### ISSUE TITLE
yt-dl add link as tag

### ISSUE THREAD

**Issue by horusra** [ytdl-org/youtube-dl#29493]
Is it possible to have yt-dl automatically write the source link (that you passed to it, to download from) to the video's ID-tag?
That way if you want to remember where you got a file from, you can just check the tags. So the source/origin will remain with the file.

**Comment by rautamiekka** [comment]
Not a bad idea, but you should always have the website-specific ID in the filename for finding dupes much faster, anyway.

**Comment by xenova** [comment]
Does `--add-metadata` not do this?

**Comment by kikuyan** [comment]
If you use `--xattrs`, url is written to `user.xdg.referrer.url` extended attr.

**Comment by horusra** [comment]
@xenova  
> Does `--add-metadata` not do this?
no it does not.

**Comment by kikuyan** [comment]
Sorry, I forgot about this:
On macOS or Linux, you need to have xattr or pyxattr python module, or on Linux, setfattr command (On macOS, in case you run youtube-dl with pre-installed python 2.7, no need to add anything.)
On Windows and NTFS, metadata seem to be written to Alternate data stream (ADS), but I'm not sure about this.

If the file is mp4, url is not added with `--add-metadata` (mp4 doesn't have `purl` metadata).

**Comment by horusra** [comment]
@kikuyan 
you're right, it seems my earlier report about it working for youtube was because it was saving those to MKV instead of MP4 since I have `--add-metadata` in youtube-dl.conf

Is there a workaround for this? can we not force the source URL to be written to something that MP4 does support? Such as `comment` `label` or `copyright` metadata tags? Depending on what the user configures?
